modules/post/linux/manage/busybox_control.rb

```diff
@@ -0,0 +1,76 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Post
+
+  def initialize
+    super(
+      'Name'         => 'BusyBox Remote Control ',
+      'Description'  => 'This module will send a script to a open session
+                         that is connected to a BusyBox sh shell. The script
+                         will accept some commands to control the target
+                         router or device executing BusyBox. Once the
+                         script is executed it will accept commands, use
+                         the help command to list the options of the script',
+      'Author'       => 'Javier Vicente Vallejo',
+      'License'      => MSF_LICENSE,
+      'References'   =>
+        [
+          [ 'URL', 'http://vallejo.cc']
+        ],
+      'Platform'      => ['linux'],
+      'SessionTypes'  => ['shell']
+    )
+  end
+
+
+
+  def run
+
+
+    file = ::File.join(Msf::Config.data_directory, "post", "busybox_control.sh")
```

Comment:
Honestly... I'm not really comfortable with this way of writing a post module. I mean as a shell script. I feel like most or some of the features provided by the busybox_control.sh could be written with help of the `Msf::Post` API.  I'm not sure if you wrote it in this way because you were having too much problems using the `Msf::Post`` methods from a busybox session :?  But definitely my first feeling is which this busybox_control.sh should be rewritten in ruby.